docs/setup/custom-config.mdx

```diff
@@ -44,40 +39,99 @@ python -m mindsdb --no_studio
 
 ## Starting MindsDB with Extended Configuration
 
-First, you should prepare a `config.json` file based on the following template; remember to substitute the values for your custom configuration.
+Start MindsDB locally with your custom configuration by providing a path to the `config.json` file that stores custom config parameters listed in this section.
+
+```bash
+python -m mindsdb --config=/path-to-the-extended-config-file/config.json
+```
+
+Below are all of the custom configuration parameters that should be set according to your requirements and saved into the `config.json` file.
 
 ```bash
 {
     "permanent_storage": {
         "location": "local"
+        "bucket": "s3_bucket_name" # optional
     },
+```
+
+The `permanent_storage` parameter defines where MindsDB stores essential user files, such as uploaded files, models, and tab content, providing an active storage mechanism. MindsDB checks the `permanent_storage` location to access the latest version of a file and updates it as needed.
+
+The `location` specifies the storage type.
+
+* `absent` (default): Disables permanent storage and is not recommended to use.
+* `local`: Stores files in a local directory defined with `config['paths']['storage']`.
+* `s3`: Stores files in an Amazon S3 bucket. This option requires the `bucket` parameter that specifies the name of the S3 bucket where files will be stored.
+
+```bash
     "storage_dir": "/home/mindsdb/var",
+```
+
+The `storage_dir` parameter specifies the directory where MindsDB stores all its files, including models, uploaded files, tab content, and the internal SQLite database (if running locally).
+
+If this parameter is not set, the path is determined by the `MINDSDB_STORAGE_DIR` environment variable. MindsDB defaults to creating a `mindsdb` folder in the operating system user's home directory.
+
+```bash
     "paths": {
         "root": "/home/mindsdb/var",
```

Comment:
`root` is also optional, it can be defined with `MINDSDB_STORAGE_DIR` or it will be in home dir by default